### ISSUE TITLE
Qwen3.5-35B-A3B val loss nan

### ISSUE THREAD

**Issue by jalola** [NVIDIA-NeMo/Automodel#1455]
**Describe the bug**

Qwen3.5-35B-A3B val loss is always nan although train loss is correct

**Steps/Code to reproduce bug**

Train with the model [qwen3_5_35b.yaml](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/vlm_finetune/qwen3_5_moe/qwen3_5_35b.yaml)

I used `val_every_steps: 4` for quick debugging

Log:

```
2026-03-04 18:53:48 | INFO | root | step 0 | epoch 0 | loss 2.0552 | grad_norm 16.2790 | lr 5.00e-06 | mem 40.47 GiB | tps 26.97(3.37/gpu) | num_label_tokens 1686
2026-03-04 18:53:53 | INFO | root | step 1 | epoch 0 | loss 1.7913 | grad_norm 10.0579 | lr 5.00e-06 | mem 56.27 GiB | tps 1193.45(149.18/gpu) | num_label_tokens 1706
2026-03-04 18:53:57 | INFO | root | step 2 | epoch 0 | loss 1.9760 | grad_norm 7.5852 | lr 5.00e-06 | mem 56.31 GiB | tps 1179.93(147.49/gpu) | num_label_tokens 1348
2026-03-04 18:54:03 | INFO | root | step 3 | epoch 0 | loss 1.2682 | grad_norm 6.1952 | lr 5.00e-06 | mem 56.39 GiB | tps 937.45(117.18/gpu) | num_label_tokens 1309
2026-03-04 18:59:15 | INFO | root | [val] step 3 | epoch 0 | loss nan | lr 5.00e-06 | num_label_tokens 279002
2026-03-04 18:59:20 | INFO | root | step 4 | epoch 0 | loss 1.8212 | grad_norm 9.5968 | lr 5.00e-06 | mem 56.23 GiB | tps 14.51(1.81/gpu) | num_label_tokens 790
2026-03-04 18:59:24 | INFO | root | step 5 | epoch 0 | loss 1.6419 | grad_norm 12.5712 | lr 5.00e-06 | mem 56.24 GiB | tps 1103.71(137.96/gpu) | num_label_tokens 442
2026-03-04 18:59:28 | INFO | root | step 6 | epoch 0 | loss 1.8354 | grad_norm 5.7311 | lr 5.00e-06 | mem 56.35 GiB | tps 1433.06(179.13/gpu) | num_label_tokens 1682
2026-03-04 18:59:32 | INFO | root | step 7 | epoch 0 | loss 1.8029 | grad_norm 8.3803 | lr 5.00e-06 | mem 56.33 GiB | tps 1105.92(138.24/gpu) | num_label_tokens 945
2026-03-04 19:04:41 | INFO | root | [val] step 7 | epoch 0 | loss nan | lr 5.00e-06 | num_label_tokens 279002
```

**Expected behavior**

val loss should not be nan

**Additional context**

I tried with another dataset, same issue

**Comment by qiaochuz-nv** [comment]
## QA RCCA Analysis

**Fix PR:** #1428
**Root Cause:** Qwen3.5-35B-A3B validation loss always NaN — MoE validation step had incorrect loss computation, dividing by wrong token count.

**Trigger Config:** Qwen3.5 MoE + validation step

**Coverage Assessment:**

| Dimension | Bug Config | Nearest Test | Match |
|-----------|-----------|--------------|-------|
| Qwen3.5 MoE | Required | None (only dense Qwen3.5 test) | No |
| Validation step | Required | Tests with val exist but not MoE | Partial |
| Loss correctness | Silent bug | No assertion on loss values | No |

**Status:** PARTIAL — only dense Qwen3.5 test exists. MoE variant untested. This was a silent correctness issue (NaN loss, not a crash), making it harder to catch with smoke tests.

**Conclusion:** MoE validation correctness is a known gap. The new `automodel_qwen3_5_moe_lora_ac_ckpt_test.sh` (pending validation) covers Qwen3.5 MoE training but validation-specific assertions remain a gap for silent correctness bugs.
